integration-tests/pipelines/e2e-tests-staging-pipeline.yaml

```diff
@@ -0,0 +1,153 @@
+        - name: KUBECONFIG_SECRET_NAME
+          value: $(params.KUBECONFIG_SECRET_NAME)
+
+      taskSpec:
+        params:
+          - name: PIPELINE_TEST_SUITE
+          - name: STEP_IMAGE
+          - name: PR_GIT_URL
+          - name: PR_GIT_REVISION
+          - name: VAULT_PASSWORD_SECRET_NAME
+          - name: GITHUB_TOKEN_SECRET_NAME
+        results:
+          - name: TEST_OUTPUT
+            description: Test output
+        steps:
+          - name: run-test
+            image: $(params.STEP_IMAGE)
+            env:
+              - name: SNAPSHOT
+                value: $(params.SNAPSHOT)
+              - name: VAULT_PASSWORD
+                valueFrom:
+                  secretKeyRef:
+                    name: $(params.VAULT_PASSWORD_SECRET_NAME)
+                    key: password
+              - name: GITHUB_TOKEN
+                valueFrom:
+                  secretKeyRef:
+                    name: $(params.GITHUB_TOKEN_SECRET_NAME)
+                    key: token
+              - name: KUBECONFIG
+                valueFrom:
+                  secretKeyRef:
+                    name: $(params.KUBECONFIG_SECRET_NAME)
+                    key: kubeconfig
+            script: |
+              #!/usr/bin/env bash
+
+              exitfunc() {
+                local err=$1
+                local line=$2
+                local command="$3"
+
+                successcount=0
+                failurecount=0
+                if [ "$err" -eq 0 ] ; then   
+                  # After the tests finish, record the overall result in the RESULT variable
+                  RESULT="SUCCESS"
+                  successcount=1
+                else
+                  echo -n \
+                  "$0: ERROR $command failed at line $line - exited with status $err"
+                  RESULT="FAILURE"
+                  failurecount=1
+                fi
+                # Output the standardized TEST_OUTPUT result in JSON form
+                TEST_OUTPUT=$(jq -rc --arg date "$(date -u --iso-8601=seconds)" --arg RESULT "${RESULT}" \
+                  --argjson successcount ${successcount} --argjson failurecount ${failurecount} --null-input \
+                  '{result: $RESULT, timestamp: $date, failures: $failurecount, successes: $successcount, warnings: 0}')
+                echo -n "${TEST_OUTPUT}" | tee "$(results.TEST_OUTPUT.path)"
+                exit 0 # exit the script cleanly as there is no point in proceeding past an error or exit call
+              }
+              # due to set -e, this catches all EXIT and ERR calls and the task should never fail with nonzero exit code
+              trap 'exitfunc $? $LINENO "$BASH_COMMAND"' EXIT
+
+              echo "PR_GIT_URL: $(params.PR_GIT_URL)"
+              echo "PR_GIT_REVISION: $(params.PR_GIT_REVISION)"
+
+              VAULT_PASSWORD_FILE=$(mktemp)
+              export VAULT_PASSWORD_FILE
+              set +x
+              echo "${VAULT_PASSWORD:?}" > "${VAULT_PASSWORD_FILE}"
```

Comment:
just FYI - no change needed - if you use `cat` instead of `echo`:
```
cat > ${VAULT_PASSWORD_FILE} <<EOF
${VAULT_PASSWORD}
EOF
```
It won't  leak the `$VAULT_PASSWORD` even with `-x` set